src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2307 @@
+        if (i_nuclide >= 0) {
+          score = get_nuclide_xs(i_nuclide+1, MG_GET_XS_NU_FISSION, p_g)
+            * atom_density * flux;
+        } else {
+          score = get_macro_xs(p->material, MG_GET_XS_NU_FISSION, p_g) * flux;
+        }
+      }
+      break;
+
+
+    case SCORE_PROMPT_NU_FISSION:
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        if (settings::survival_biasing || p->fission) {
+          if (tally.energyout_filter_ != C_NONE) {
+            // Fission has multiple outgoing neutrons so this helper function
+            // is used to handle scoring the multiple filter bins.
+            score_fission_eout(p, i_tally, score_index, score_bin);
+            continue;
+          }
+        }
+        if (settings::survival_biasing) {
+          // No fission events occur if survival biasing is on -- need to
+          // calculate fraction of absorptions that would have resulted in
+          // prompt-nu-fission
+          score = p->absorb_wgt * flux;
+          if (i_nuclide >= 0) {
+            score *= atom_density
+              * get_nuclide_xs(i_nuclide+1, MG_GET_XS_PROMPT_NU_FISSION, p_g)
+              / get_macro_xs(p->material, MG_GET_XS_ABSORPTION, p_g);
+          } else {
+            score *=
+              get_macro_xs(p->material, MG_GET_XS_PROMPT_NU_FISSION, p_g)
+              / get_macro_xs(p->material, MG_GET_XS_ABSORPTION, p_g);
+          }
+        } else {
+          // Skip any non-fission events
+          if (!p->fission) continue;
+          // If there is no outgoing energy filter, than we only need to score
+          // to one bin. For the score to be 'analog', we need to score the
+          // number of particles that were banked in the fission bank. Since
+          // this was weighted by 1/keff, we multiply by keff to get the proper
+          // score.
+          auto n_delayed = std::accumulate(p->n_delayed_bank,
+            p->n_delayed_bank+MAX_DELAYED_GROUPS, 0);
+          auto prompt_frac = 1. - n_delayed / static_cast<double>(p->n_bank);
+          score = simulation::keff * p->wgt_bank * prompt_frac * flux;
+          if (i_nuclide >= 0) {
+            score *= atom_density
+              * get_nuclide_xs(i_nuclide+1, MG_GET_XS_FISSION, p_g)
+              / get_macro_xs(p->material, MG_GET_XS_FISSION, p_g);
+          }
+        }
+      } else {
+        if (i_nuclide >= 0) {
+          score = get_nuclide_xs(i_nuclide+1, MG_GET_XS_PROMPT_NU_FISSION, p_g)
+            * atom_density * flux;
+        } else {
+          score = get_macro_xs(p->material, MG_GET_XS_PROMPT_NU_FISSION, p_g)
+            * flux;
+        }
+      }
+      break;
+
+
+    case SCORE_DELAYED_NU_FISSION:
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        if (settings::survival_biasing || p->fission) {
+          if (tally.energyout_filter_ != C_NONE) {
+            // Fission has multiple outgoing neutrons so this helper function
+            // is used to handle sco
```

Comment:
I'm not sure what you mean.  Can you explain?

Reply:
Down in the for loop, the condition to be checked is `filter_iter != end` so the loop will exit if they're not equal

Reply:
Oh, right.  Unfortunately, the `EnergyoutFilter::get_all_bins` doesn't bin outgoing fission neutrons.  It wouldn't be easy to change that either because then the filter would need to be specific for the current `score_bin`

Reply:
Wait, I'm not sure if we're talking about the same thing here. On the github page, it looks like my comment is showing up attached to the wrong line. I'm talking about tally_scoring.cpp:1990

Reply:
Oh, wild.  Yeah, it's showing up in the correct spot if I look at the "Files changed" tab, but not on the "Conversation" tab.  That `continue` is needed to ensure we don't hit the `assume_separate` check.

Reply:
Ah, gotcha. Thanks!